device.go

```diff
@@ -15,41 +15,59 @@ type InputDevice struct {
 	driverVersion int32
 }
 
-// Open creates a new InputDevice from the given path. Returns an error if
+// OpenWithFlag creates a new InputDevice from the given path. The input device
+// is opened with the specified flag (O_RDONLY etc.). Returns an error if
 // the device node could not be opened or its properties failed to read.
-func Open(path string) (*InputDevice, error) {
+func OpenWithFlag(path string, flag int) (*InputDevice, error) {
```

Comment:
One thought here is that this would be better named `OpenWithFlags()` (plural) as users are permitted to submit a mask with many flags set. I know that the parameter of `os.Open()` is called `flag`, but I find this equally confusing.

Another concern is that this would now allow to pass flags such as `os.O_CREATE`, `os.O_APPEND`, `os.O_TRUNC` etc, which make no sense in this context. However, things such as `O_CLOEXEC` are very useful. But then again, we can make it the user's responsibility to use this interface sanely. Maybe just add a comment on that?

Reply:
Thanks for the good feedback.

> would be better named `OpenWithFlags()` (plural) 

Good point, will change it.

> Another concern is that this would now allow to pass flags such as os.O_CREATE, os.O_APPEND, os.O_TRUNC etc, which make no sense in this context. 

Generally, I agree. But shouldn't it be the kernel's responsibility to avoid misuse such that `os.OpenFile()` returns an error?

Leaving it to the user and adding a comment sounds good to me. Looking at what what others are doing we are in good companion: [evdev-rs](https://ndesh26.github.io/evdev-rs/evdev_rs/struct.Device.html) and [libevdev](https://www.freedesktop.org/software/libevdev/doc/latest/) even leave opening the fd to the caller.

Reply:
> Leaving it to the user and adding a comment sounds good to me. Looking at what what others are doing we are in good companion: [evdev-rs](https://ndesh26.github.io/evdev-rs/evdev_rs/struct.Device.html) and [libevdev](https://www.freedesktop.org/software/libevdev/doc/latest/) even leave opening the fd to the caller.

Yes, then let's do that.

> Good point, will change it.

Great. Let me know when you did that, then we can merge.

Reply:
Done